File: frontend/src/api/email.ts
import api from './index'

export interface EmailAccount {
  id: number
  name: string
  email: string
  imap_server: string
  smtp_server: string
  is_active: boolean
  created_at: string
}

export interface EmailMessage {
  id: number
  account_id: number
  subject: string
  sender: string
  recipients: string
  body_text?: string
  body_html?: string
  is_read: boolean
  is_starred: boolean
  has_attachment: boolean
  received_at?: string
  created_at: string
}

export interface SendMessageParams {
  to: string[]
  subject: string
  body: string
  html?: boolean
}

export const emailApi = {
  listAccounts() {
    return api.get<any, EmailAccount[]>('/api/email/accounts')
  },
  createAccount(data: {
    name: string
    email: string
    password: string
    imap_server: string
    imap_port: number
    smtp_server: string
    smtp_port: number
    use_ssl?: boolean
  }) {
    return api.post<any, EmailAccount>('/api/email/accounts', data)
  },
  deleteAccount(id: number) {
    return api.delete(`/api/email/accounts/${id}`)
  },
  getUnreadCount(accountId: number) {
    return api.get<any, { account_id: number; folder: string; unread_count: number }>(`/api/email/accounts/${accountId}/unread`)
  },
  listMessages(accountId: number, folder?: string) {
    return api.get<any, { total: number; items: EmailMessage[] }>(`/api/email/accounts/${accountId}/messages`, { params: { folder } })
  },
  getMessage(messageId: number) {
    return api.get<any, EmailMessage>(`/api/email/messages/${messageId}`)
  },
  sendMessage(accountId: number, data: SendMessageParams) {
    return api.post(`/api/email/accounts/${accountId}/send`, data)
  },
  markAsRead(messageId: number) {
    return api.put(`/api/email/messages/${messageId}/read`)
  },
  toggleStar(messageId: number) {
    return api.put(`/api/email/messages/${messageId}/star`)
  },
  syncEmails(accountId: number) {
    return api.post(`/api/email/accounts/${accountId}/sync`)
  }
}

File: backend/app/schemas/email.py
"""Email module schemas."""

from pydantic import BaseModel, EmailStr
from datetime import datetime
from typing import Optional, List


class EmailAccountCreate(BaseModel):
    name: str
    email: str
    imap_server: str
    imap_port: int = 993
    imap_ssl: bool = True
    smtp_server: str
    smtp_port: int = 465
    smtp_ssl: bool = True
    username: str
    password: str  # 授权码/密码，存储时加密


class EmailAccountUpdate(BaseModel):
    name: Optional[str] = None
    imap_server: Optional[str] = None
    imap_port: Optional[int] = None
    imap_ssl: Optional[bool] = None
    smtp_server: Optional[str] = None
    smtp_port: Optional[int] = None
    smtp_ssl: Optional[bool] = None
    username: Optional[str] = None
    password: Optional[str] = None
    is_active: Optional[bool] = None


class EmailAccountResponse(BaseModel):
    id: int
    name: str
    email: str
    imap_server: str
    smtp_server: str
    is_active: bool
    created_at: datetime

    class Config:
        from_attributes = True


class EmailMessageResponse(BaseModel):
    id: int
    account_id: int
    subject: Optional[str]
    sender: Optional[str]
    recipients: Optional[str]
    body_text: Optional[str]
    is_read: bool
    is_starred: bool
    has_attachment: bool
    folder: Optional[str]
    received_at: Optional[datetime]

    class Config:
        from_attributes = True


class EmailMessageListResponse(BaseModel):
    total: int
    items: List[EmailMessageResponse]


class SendEmailRequest(BaseModel):
    to: List[str]
    subject: str
    body: str
    html: bool = False
    cc: Optional[List[str]] = None
